### ISSUE TITLE
Question: why is ssl used in a 'unverified' mode ?

### ISSUE THREAD

**Issue by dmerejkowsky** [timonweb/pytailwindcss#1]
The code that fetches the tailwindcss binaries from github looks like this:

```python
    # Ensures SSL certificate verification doesn't fail
    ssl._create_default_https_context = ssl._create_unverified_context
    urllib.request.urlretrieve(url, str(dest_filename))
```

This is called by the `main()` function.

This means anyone using `pytailwindcss` after installation will try and download binaries from `github.com`, *but* without making sure the binary files come from the correct host, which can lead to all kinds of attack (and defeats the purpose of using SSL in the first place).

If they are legitimate cases for the SSL verification to be skipped, I think the code above can be kept, but *only* if something like an `--insecure` option is passed. By the way, this is what both `curl` and `wget` do.

Feel free to tell me if I've missed something or if you need more information. I can also work on a patch if you like.

**Comment by timonweb** [comment]
This was a quick solution to solve SSL verification issues on Mac OS. I agree, it's ugly, so if you have a better idea, feel free to submit a PR. Thanks!

**Comment by dmerejkowsky** [comment]
> to solve SSL verification issues on Mac OS. 

What kind of issues? I think it would be wise to know more before making a patch. There may be a better way to solve the issue that does not create a security risk for all users.

**Comment by dmerejkowsky** [comment]
Actually, after thinking about it a bit more  I *do* have a solution.

Rather than having `pytailwindcss` download the binary from github, build a wheel for each platform containing the correct binary, and then let `pip` fetch the correct binary from `pypi.org`.

This is what is done for the `cmake` package on pypi, by the way.

**Comment by timonweb** [comment]
Like outdated certificates that come with a Python version, here is a good description of the issue https://sectigostore.com/page/what-is-an-ssl-certificate_verify_failed-error-and-how-do-i-resolve-it/

Potentially, we could solve this by using curl via subprocess call.

**Comment by timonweb** [comment]
> Actually, after thinking about it a bit more  I *do* have a solution.
> 
> 
> 
> Rather than having `pytailwindcss` download the binary from github, build a wheel for each platform containing the correct binary, and then let `pip` fetch the correct binary from `pypi.org`.
> 
> 
> 
> This is what is done for the `cmake` package on pypi, by the way.

Oh no, I don't have time and resources for something like that. This package is supposed to be low-maintenance and grab binaries created by Tailwind CSS team.

**Comment by laundmo** [comment]
Personally, i think the fact that this downloads binaries without ssl verification makes it unusable. It would be quite easy to man-in-the-middle the binaries to something malicious.

**Comment by timonweb** [comment]
@dmerejkowsky @laundmo thank you for raising my awareness. I just removed that escape hatch. If someone has outdated SSL certs on his system, it's their problem not the majority of users, so I agree, making sure that SSL is in place has a higher value than making sure it downloads a binary on any machine. I released version `0.1.1` of the package: https://pypi.org/project/pytailwindcss/0.1.1/
